api/src/team/scripts/send-organization-invitations.script.js

```diff
@@ -54,24 +56,47 @@ export class SendOrganizationInvitationsScript extends Script {
         },
       },
     });
+    this.failedLines = 0;
+    this.successLines = 0;
+    this.actualLine = 0;
   }
 
-  async handle({ options, logger, sendOrganizationInvitation = usecases.createOrganizationInvitationByAdmin }) {
+  async handle({
+    options,
+    logger,
+    sendOrganizationInvitation = usecases.createOrganizationInvitationByAdmin,
+    checkMail = checkDomainIsValid,
+  }) {
     const { file, dryRun, batchSize, throttleDelay } = options;
     const batches = _.chunk(file, batchSize);
     let batchCount = 1;
     for (const batch of batches) {
       logger.info(`Batch #${batchCount++}`);
-      batch.map(async (invitation) => {
-        if (!dryRun) {
-          await sendOrganizationInvitation({
-            organizationId: invitation['Organization ID'],
-            email: invitation.email,
-            locale: invitation.locale,
-            role: invitation.role || null,
-          });
-        }
-      });
+      Promise.all(
+        batch.map(async (invitation) => {
+          if (!dryRun) {
+            this.actualLine++;
+            try {
+              const checking = await checkMail(invitation.email);
+              if (checking) {
```

Comment:
Attention, en fait `checkDomainIsValid` est une assertion, c'est à dire qu'il ne faut pas attendre un résultat booléen de son exécution. Si cela est une réussite la promesse est résolue, sinon ça lance une exception.

```suggestion
              await checkDomainIsValid(invitation.email);
```

Reply:
En fait `checkDomainIsValid` devrait être nommé `assertDomainIsValid`. Je vais faire cela dans une autre PR que je créerai après que cette PR-ci soit validée.